src/overlays/actors/ovl_En_Look_Nuts/z_en_look_nuts.c

```diff
@@ -71,34 +92,299 @@ static DamageTable D_80A68630 = {
+    if (D_80A6862C == 0) {
+        if ((this->state < 2) && (this->actor.xzDistToPlayer < 320.0f) && (this->actor.playerHeightRel < 80.0f)) {
+            effectVelOffset = effectVecInitialize;
+            Math_Vec3f_Copy(&effectPos, &this->actor.world.pos);
+            effectPos.x += Math_SinS((this->actor.world.rot.y + (s16)this->headRotation.y)) * 10.0f;
+            effectPos.y += 30.0f;
+            effectPos.z += Math_CosS((this->actor.world.rot.y + (s16)this->headRotation.y)) * 10.0f;
+            Matrix_StatePush();
+            Matrix_RotateY(this->actor.shape.rot.y, MTXMODE_NEW);
+            effectVelOffset.z = 20.0f;
+            Matrix_MultiplyVector3fByState(&effectVelOffset, &effectVel);
+            Matrix_StatePop();
+            if (this->isPlayerDetected == false) {
+                s16 drawFlag = 1;
+                if (gSaveContext.isNight) {
+                    drawFlag = 0;
+                }
+                if (Player_GetMask(globalCtx) != PLAYER_MASK_STONE) {
+                    EffectSsSolderSrchBall_Spawn(globalCtx, &effectPos, &effectVel, &gZeroVec3f, 50,
+                                                 &this->isPlayerDetected, drawFlag);
+                }
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Look_Nuts/func_80A67D0C.s")
+            if ((this->isPlayerDetected == true) || (this->actor.xzDistToPlayer < 20.0f)) {
+                Player* player = GET_PLAYER(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Look_Nuts/func_80A67F30.s")
+                if (!(player->stateFlags3 & 0x100) && !func_801690CC(globalCtx)) {
+                    Math_Vec3f_Copy(&this->headRotationTarget, &gZeroVec3f);
+                    this->state = PALACE_GUARD_RUNNING_TO_PLAYER;
+                    play_sound(NA_SE_SY_FOUND);
+                    func_800B7298(globalCtx, &this->actor, 0x1A);
+                    D_80A6862C = 1;
+                    this->actor.flags |= (ACTOR_FLAG_1 | ACTOR_FLAG_10);
+                    this->actor.gravity = 0.0f;
+                    EnLookNuts_DetectedPlayer(this, globalCtx);
+                } else {
+                    this->isPlayerDetected = false;
+                }
+            }
+        }
+        Math_ApproachF(&this->headRotation.x, this->headRotationTarget.x, 1.0f, 3000.0f);
+        Math_ApproachF(&this->headRotation.y, this->headRotationTarget.y, 1.0f, 6000.0f);
+        Math_ApproachF(&this->headRotation.z, this->headRotationTarget.z, 1.0f, 2000.0f);
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Look_Nuts/func_80A67FC4.s")
+static TexturePtr sEyeTextures[] = { gDekuPalaceGuardEyeOpenTex, gDekuPalaceGuardEyeHalfTex,
+                                     gDekuPalaceGuardEyeClosedTex };
```

Comment:
```suggestion
static TexturePtr sEyeTextures[] = { gDekuPalaceGuardEyeOpenTex, gDekuPalaceGuardEyeHalfTex,
                                     gDekuPalaceGuardEyeClosedTex, };
```
Probably formats better with trailing comma